### PR TITLE
[DataGrid] Allow to customize the value displayed in the filter button tooltip

### PR DESCRIPTION
**Pull request by ithrforu**
Related to https://github.com/mui/mui-x/issues/5492 (a follow-up is to fix the built-in operators)

I faced a problem with current filter values in GridToolbarFilterButton tooltip title (use DataGridPremium) and found [issue](https://github.com/mui/mui-x/issues/5492#issuecomment-1185835877) with @m4theushw comment. This problem looks like:
image
image

In this PR i added the ability (from @m4theushw proposal) to specify a function to format the filter value in GridToolbarFilterButton tooltip title.

### CODE REVIEW

File: packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx [hunk 1]
```diff
@@ -78,6 +78,14 @@ const GridToolbarFilterButton = React.forwardRef<HTMLButtonElement, GridToolbarF
           .getLocaleText(`filterOperator${capitalize(item.operatorValue!)}` as GridTranslationKeys)!
           .toString();
 
+      const getFilterItemValue = (item: GridFilterItem): string => {
+        const { getValueAsString } = lookup[item.columnField!].filterOperators!.find(
+          (operator) => operator.value === item.operatorValue,
+        )!;
+
+        return getValueAsString ? getValueAsString(item.value) : item.value;
+      };
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx:87]: Filter item got [updated](https://github.com/mui/mui-x/pull/6985), can you rebase with latest `next` ?
```suggestion
      const getFilterItemValue = (item: GridFilterItem): string => {
        const { getValueAsString } = lookup[item.field!].filterOperators!.find(
          (operator) => operator.value === item.operator,
        )!;

        return getValueAsString ? getValueAsString(item.value) : item.value;
      };
```

Comment (ithrforu) [reply packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx]: All done.

**Comment by MBilalShafi** [comment]
@ithrforu Thanks for the contribution, it seems good, can we test the newly added function `getValueAsString` and add some example usage in demos to improve usability?

**Comment by ithrforu** [comment]
@MBilalShafi. Thank you for attention. Of course. Should I create new pull request or continue here?

**Comment by MBilalShafi** [comment]
> Should I create new pull request or continue here

It will be better if we have it as part of this pull request so that we push this new addition tested and more usable.

**Comment by ithrforu** [comment]
@MBilalShafi, I added an example and rebased with latest next, but got error in [docs](https://deploy-preview-6956--material-ui-x.netlify.app/x/react-data-grid/filtering/) page.
image

How can i test `getValueAsString`?

**Comment by MBilalShafi** [comment]
> How can I test `getValueAsString`?

You can create a new test case in https://github.com/mui/mui-x/blob/next/packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx covering this use-case and test if it works as expected.

**Comment by ithrforu** [comment]
@MBilalShafi, all done.

File: docs/data/data-grid/filtering/filtering.md [hunk 2]
```diff
@@ -203,6 +203,10 @@ The [`valueFormatter`](/x/react-data-grid/column-definition/#value-formatter) is
 If the column has a [`valueGetter`](/x/react-data-grid/column-definition/#value-getter), then `params.value` will be the resolved value.
 :::
 
+:::info
+The `getValueAsString` function allows you to convert the value of a filter item to a human-readable form before displayed in the UI of filter button tooltip.
```
Comment (MBilalShafi) [inline docs/data/data-grid/filtering/filtering.md:207]:
```suggestion
The filter button displays a tooltip on hover if there are active filters. Pass [`getValueAsString`](/x/api/data-grid/grid-filter-operator/) in the filter operator to customize or convert the value to a more human-readable form.
```

Comment (ithrforu) [reply docs/data/data-grid/filtering/filtering.md]: Thanks for your edits and remarks. Text is fixed.

**Review by MBilalShafi — APPROVED** [review]

**Comment by ithrforu** [comment]
Is it bug?
MUI

**Comment by ithrforu** [comment]
@m4theushw, I mean I hovered filter button but "Show the code" tooltip displayed.

**Comment by LukasTy** [comment]
Indeed, this PR might have some sort of regression in hover event bubbling or something along these lines. The current versions do not display `Show the full source` tooltip on demo hover. 🙈

**Comment by LukasTy** [comment]
> I can also reproduce this bug in https://material-ui.netlify.app/material-ui/react-button/ so certainly is not a regression introduced by this PR but from the core.

Hm. Good point. I tried testing a local dev run and did not see this behaviour, that's why I thought that it might have something to do with this PR, but I do not see any `@mui/monorepo` bumps to warrant that. 🙈

**Comment by ithrforu** [comment]
Can i do something?

**Comment by MBilalShafi** [comment]
It's all good @ithrforu, I'll merge this now.
Regarding the above-mentioned regression, I'll create a ticket in core if there's not one already.

**Comment by LukasTy** [comment]
> It's all good @ithrforu, I'll merge this now. Regarding the above-mentioned regression, I'll create a ticket in core if there's not one already.

It might have also been a desired change from MUI Core, we just weren't aware of it. But I couldn't find a recently merged PR related to it. 🤔

**Pull request merged**